nselib/httpcookies.lua

```diff
@@ -0,0 +1,465 @@
+    options.cookies = self.check_cookie_attributes(self, host, port, path)
+    response = http.generic_request(host, port, method, path, options)
+    if response and response.status == 200 and response.cookies then
+      response.cookies = self.merge_cookie_table(self, response.cookies)
+    end
+    return response
+  end,
+
+  ---This function servers as an easy method to add cookies to the existing cookie jar.
+  --We can use this function to add arbitary cookie attributes with ease from our scripts
+  --@param cookie_table A cookie table to be added to existing cookies. 
+  add_cookie = function(self, cookie_table)
+    local status
+    status = self.parse(cookie_table)
+    local cookies = {}
+    table.insert(cookies, cookie_table)
+    if status then 
+      self.merge_cookie_table(self, cookies)
+      return true
+    end
+    return false
+   end,
+
+  ---This function can be used to update a cookie with a different value.
+  --@param cookie_table A cookie table where cookie_table.name matches the name of the cookie the 
+  --value of which has to be updated. 
+  --@return status Returns true if cookie is present and successfully removed.
+  update_cookie = function(self, cookie_table)
+    local status
+    status = self.parse(cookie_table)
+    if status == false then
+      return false
+    end
+    for index, cookie in pairs(self.cookies) do
+      if cookie.name == cookie_table.name then
+        self.cookies[index] = cookie_table
+        return true
+      end
+    end
+    return false
+  end,
+
+  ---This function can be used to delete a particular cookie from the cookie jar.
+  --@param cookie_name A cookie name which has to be deleted from the cookie jar. 
+  --@return status Returns true if cookie is present and successfully removed.
+  delete_cookie = function (self, cookie_name)
+    for index, cookie in pairs(self.cookies) do
+      if cookie.name == cookie_name then
+        self.cookies[index] = nil
+        return true
+      end
+    end
+    return false
+  end,
+
+  ---This function can be used to get the value of the cookie 
+  --@param cookie_name A cookie table for which the value will be returned. 
+  --@return status If cookie is found, true is returned
+  get_cookie = function (self, cookie_name)
+    for index, cookie in pairs(self.cookies) do 
+      if cookie.name == cookie_name then
+        return true, self.cookies[index]
+      end
+    end
+    return false
+  end,
+
+}
+
+if not unittest.testing() then
+  return _ENV
+end
+
+test_suite = unittest.TestSuite:new()
+
+do
+  local cookie1 = {
+      name = "SESSIONID",
+      value = "IgAAABjN8b3xxx",
+      secure = true
+  }
+  
+  local cookie2 = {
+      name = "SID",
+      value = "low",
+      ["max-age"] = "1200",
+  }
+
+  local cookie3 = {
+      name = "session_id",
+      value = "76ca8bc8c19"
+  }
+
+  local cookiejar = {}
+
+  table.insert(cookiejar, cookie1)
+  table.insert(cookiejar, cookie1)
```

Comment:
Are you sure you want to insert the same cookie 3 times?
Shouldn't you be inserting the three different cookies you created?